robot/impl/resource_manager.go

```diff
@@ -599,6 +599,16 @@ func (manager *resourceManager) Close(ctx context.Context) error {
 	return allErrs
 }
 
+// Kill attempts to kill all module processes.
+func (manager *resourceManager) Kill() {
+	// TODO: Kill processes in processManager as well.
+
+	// moduleManager may be nil in tests
+	if manager.moduleManager != nil {
```

Comment:
I suspect this can be a data race as well?

Reply:
yep! added some locking around this